libgadget/cooling.c

```diff
@@ -790,40 +789,6 @@ void MakeCoolingTable(void)
 
 }
 
-static double * h5readdouble(char * filename, char * dataset, int * Nread) {
-    void * buffer;
-    int N;
-    if(ThisTask == 0) {
-        BigFile bf[1];
-        big_file_open(bf, filename);
-        BigBlock bb[1];
-        if(0 != big_file_open_block(bf, bb, dataset)) {
-            endrun(-1, "Cannot open %s %s\n", filename, dataset);
-        }
-
-        N = bb->size;
-        MPI_Bcast(&N, 1, MPI_INT, 0, MPI_COMM_WORLD);
-
-        BigArray array[1];
-
-        big_block_read_simple(bb, 0, N, array, "f8");
```

Comment:
I thought read_simple would have allocated the buffer automaticially.

Reply:
https://github.com/rainwoodman/bigfile/blob/master/src/bigfile.c#L774

Reply:
It does - but it allocates it with malloc, not mymalloc. All I'm doing here is avoiding the copy from the malloc heap to the mymalloc heap.

Reply:
Basically I have inlined read_simple and changed malloc to mymalloc

Reply:
Yes the copy was less than ideal -- but what was buggy?

Reply:
See below - lines 1037 - 1044 used an uninitialized variable to allocate memory.